lsp/nls/src/requests/goto.rs

```diff
@@ -113,20 +131,26 @@ pub fn handle_references(
     // Maybe the position is pointing straight at the definition already.
     // In that case, def_locs won't have the definition yet; so add it.
     if let Some(id) = server.lookup_ident_by_position(pos)? {
-        def_locs.push(id);
-        if let Some(parent) = term {
-            // If `id` is a field name in a record, we can search through cousins
-            // to find more definitions.
-            if matches!(parent.as_ref(), Term::RecRecord(..) | Term::Record(_)) {
-                let def = Def::Field {
-                    ident: id,
-                    value: None,
-                    record: parent.clone(),
-                    metadata: FieldMetadata::default(),
-                };
-                let resolver = FieldResolver::new(server);
-                let cousins = resolver.get_cousin_defs(&def);
-                def_locs.extend(cousins.into_iter().map(|(loc, _)| loc))
+        if let Some(span) = id.pos.into_opt() {
```

Comment:
Nitpick: should we combine the two `if`, to avoid one level of nesting? Such as

```rust
if let Some(span) = server.lookup_ident_by_position(pos)?.and_then(|id| id.pos.into_opt())
```

Reply:
Oh wait, now I remember why it was like this: I needed to bind the `LocIdent` because it's used a few lines down to construct the `Def::Field`.